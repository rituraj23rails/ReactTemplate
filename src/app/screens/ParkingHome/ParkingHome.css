.mainContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.headingText {
    font-size: 20px;
    font-weight: bold;
    background-color: #000;
    color: #fff;
    padding: 20px;
}

.formBox {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 2px solid #000;
    width: 500px;
    height: 500px;
}

.textInputField {
    width: 300px;
    height: 50px;
    border: 2px solid #000;
    border-radius: 2px;
    font-size: 20px;
    padding-left: 10px;
    margin-top: 50px;
}

.textInputField:focus {
    outline: none;
}

.submitButton {
    cursor: pointer;
    width: 200px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid #000;
    margin-top: 50px;
}

.submitButton:hover {
    background-color: green;
    color: #fff;
}

@media screen and (max-width: 820px) {
    .formBox {
        width: 400px;
        height: 400px;
    }

    .headingText {
        font-size: 18px;
        font-weight: bold;
    }
    
    .textInputField {
        width: 250px;
        height: 50px;
        font-size: 18px;
        padding-left: 10px;
        margin-top: 50px;
    }
    
    .submitButton {
        width: 200px;
        height: 40px;
    }

    .submitButton:hover {
        background-color: green;
        color: #fff;
    }

}

@media screen and (max-width: 480px) {
    .formBox {
        width: 300px;
        height: 300px;
    }

    .headingText {
        font-size: 15px;
        font-weight: bold;
    }
    
    .textInputField {
        width: 200px;
        height: 50px;
        font-size: 15px;
        padding-left: 10px;
        margin-top: 50px;
    }
    
    .textInputField:focus {
        outline: none;
    }
    
    .submitButton {
        width: 100px;
        height: 40px;
    }

    .submitButton:hover {
        background-color: green;
        color: #fff;
    }
}