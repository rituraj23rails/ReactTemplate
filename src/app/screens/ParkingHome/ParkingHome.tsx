import React, { Component, useRef } from 'react';
import './ParkingHome.css';
import WithRouter from '../../../WithRouter';
import { connect } from 'react-redux';
import { updateParkingData } from '../ParkingSlots/ParkingSlice';

export interface Props {
    navigate: any;
    id: string;
    updateParkingData: (parkingData: any) => void;
}

interface S {
    parkingSlots: any;
}

interface SS {
    id: any;
}

class ParkingHome extends Component<Props, S, SS> {
    carousel: any = null;
    constructor(props: any) {
        super(props);
        this.state = {
            parkingSlots: '',
        };
    };

    componentDidMount = async () => {
        this.props.updateParkingData([]);
    }

    componentWillUnmount = () => {

    }

    handleChangeTextInput = (event: any) => {
        event?.preventDefault();
        this.setState({ parkingSlots: event.target.value });
    }

    onSubmitButton = async () => {
        this.props.navigate(`/parkingSlots`, { state: { parkingSlots: this.state.parkingSlots }});
    }

    renderInputCountryForm = () => {
        return (
            <div className="formBox">
                <p className="headingText">Welcome To Parking App</p>
                <input data-testid="textInputTestID" type="text" placeholder='Enter No of Slots' className='textInputField' value={this.state.parkingSlots} onChange={(event) => this.handleChangeTextInput(event)} />
                <button data-testid="submitButtonTestID" onClick={() => this.onSubmitButton()} className="submitButton">SUBMIT</button>
            </div>
        )
    }

    render() {
        return (
            <div className='mainContainer'>
                {this.renderInputCountryForm()}
            </div>
        )
    }
};

const mapStateToProps = (state: any) => ({
    parkingData: state.parking.parkingData
});

const mapDispatchToProps = {
    updateParkingData: updateParkingData
};

export default connect(mapStateToProps, mapDispatchToProps)(WithRouter(ParkingHome));
export {ParkingHome};
