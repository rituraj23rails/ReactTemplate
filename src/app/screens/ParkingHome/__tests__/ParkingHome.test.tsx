/**
 *
 PARKING HOME TEST FILE
 * @format
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper, configure, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { ParkingHome } from '../ParkingHome';

configure({ adapter: new Adapter() });


const screenProps = {
    navigate: jest.fn(),
    id: 'ParkingHome',
    updateParkingData: jest.fn()
}

describe('Launching ParkingHome', () => {

    beforeEach(() => {
        global.fetch = jest.fn()
    })

    it('ParkingHome root file loads properly', () => {
        const tree: any = renderer.create(<ParkingHome {...screenProps} />).toJSON();
        expect(tree.children.length).toBe(1);
        expect(tree).toMatchSnapshot();
    });

    it('Test User ParkingHome Screen', () => {
        let wrapper: ShallowWrapper;
        let instance: ParkingHome;
        wrapper = shallow(<ParkingHome {...screenProps} />)
        instance = wrapper.instance() as ParkingHome;
        let inputComponent = wrapper.findWhere((node) => node.prop("data-testid") === "textInputTestID");
        inputComponent.simulate('change', { preventDefault: jest.fn(), target: { value: 'ind' } });
        let buttonComponent = wrapper.findWhere((node) => node.prop("data-testid") === "submitButtonTestID");
        buttonComponent.simulate("click");
        instance.componentWillUnmount();
    });
});