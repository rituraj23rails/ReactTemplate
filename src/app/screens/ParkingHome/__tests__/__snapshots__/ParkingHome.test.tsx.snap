// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launching ParkingHome ParkingHome root file loads properly 1`] = `
<div
  className="mainContainer"
>
  <div
    className="formBox"
  >
    <p
      className="headingText"
    >
      Welcome To Parking App
    </p>
    <input
      className="textInputField"
      data-testid="textInputTestID"
      onChange={[Function]}
      placeholder="Enter No of Slots"
      type="text"
      value=""
    />
    <button
      className="submitButton"
      data-testid="submitButtonTestID"
      onClick={[Function]}
    >
      SUBMIT
    </button>
  </div>
</div>
`;
